Tests fail on native Windows cmd

## Expected Behavior

Tests should pass

## Observed Behavior

Tests fail on Windows (10 is what I used)

## Environment Information

```
C:\Users\xmr\Desktop\nyc>npx envinfo@latest --preset nyc
npx: installed 1 in 1.173s

  System:
    OS: Windows 10 10.0.18363
    CPU: (4) x64 Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
    Memory: 6.54 GB / 8.00 GB
  Binaries:
    Node: 12.14.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    istanbul-lib-coverage: ^3.0.0 => 3.0.0
    istanbul-lib-hook: ^3.0.0 => 3.0.0
    istanbul-lib-instrument: ^4.0.0 => 4.0.0
    istanbul-lib-processinfo: ^2.0.2 => 2.0.2
    istanbul-lib-report: ^3.0.0 => 3.0.0
    istanbul-lib-source-maps: ^4.0.0 => 4.0.0
    istanbul-reports: ^3.0.0 => 3.0.0
    source-map-support: ^0.5.16 => 0.5.16

```

## Test output

<details><summary>Log</summary>

```
C:\Users\xmr\Desktop\nyc>git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

C:\Users\xmr\Desktop\nyc>node -v && npm -v
v12.14.0
6.13.4

C:\Users\xmr\Desktop\nyc>npm i
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 847 packages from 320 contributors and audited 3391 packages in 12.536s

20 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities


C:\Users\xmr\Desktop\nyc>npm t

> nyc@15.0.0 pretest C:\Users\xmr\Desktop\nyc
> npm run lint && npm run clean && npm run instrument


> nyc@15.0.0 lint C:\Users\xmr\Desktop\nyc
> standard


> nyc@15.0.0 clean C:\Users\xmr\Desktop\nyc
> node ./npm-run-clean.js


> nyc@15.0.0 instrument C:\Users\xmr\Desktop\nyc
> node ./build-self-coverage.js


> nyc@15.0.0 test C:\Users\xmr\Desktop\nyc
> tap

 PASS  test/add-all-files.js 13 OK 2s
 PASS  test/cache.js 2 OK 5s
 PASS  test/config.js 15 OK 153.719ms
 PASS  test/eager.js 6 OK 1s
 PASS  test/cwd.js 4 OK 99.999ms
 PASS  test/instrument.js 63 OK 10s
 PASS  test/issue-190.js 2 OK 10.033ms
 FAIL  test/nyc-integration.js
 ✖ should be equal


  t.equal(status, 0)
----^
  t.equal(stderr, '')
  t.matchSnapshot(stdout, 'stdout')

  test: test/nyc-integration.js can run "npm test" which directly invokes a test file
  found: 7
  wanted: 0
  compare: ===
  at:
    line: 11
    column: 5
    file: test/helpers/test-success.js
    function: testSuccess
  stack: |
    testSuccess (test/helpers/test-success.js:11:5)

 FAIL  test/nyc-integration.js
 ✖ should be equal

  t.equal(status, 0)
  t.equal(stderr, '')
----^
  t.matchSnapshot(stdout, 'stdout')
}

  test: test/nyc-integration.js can run "npm test" which directly invokes a test file
  found: "Error: Failed to replace env in config: ${APPDATA}

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:415:13

    \    at String.replace (<anonymous>)

    \    at envReplace (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:411:12)

    \    at parseField (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:389:7)

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:330:24

    \    at Array.forEach (<anonymous>)

    \    at Conf.add (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:328:23)

    \    at ConfigChain.addString (C:/Program
    Files/nodejs/node_modules/npm/node_modules/config-chain/index.js:244:8)

    \    at Conf.<anonymous> (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:316:10)

    \    at C:/Program
    Files/nodejs/node_modules/npm/node_modules/graceful-fs/graceful-fs.js:115:16

    TypeError: Cannot read property 'loaded' of undefined

    \    at exit (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97:27)

    \    at errorHandler (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:216:3)

    \    at C:/Program Files/nodejs/node_modules/npm/bin/npm-cli.js:78:20

    \    at cb (C:/Program Files/nodejs/node_modules/npm/lib/npm.js:225:22)

    \    at C:/Program Files/nodejs/node_modules/npm/lib/npm.js:263:24

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:81:7

    \    at Array.forEach (<anonymous>)

    \    at Conf.<anonymous> (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:80:13)

    \    at Conf.f (C:/Program
    Files/nodejs/node_modules/npm/node_modules/once/once.js:25:25)

    \    at Conf.emit (events.js:210:5)

    C:/Program Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97\r

    \  var doExit = npm.config.loaded ? npm.config.get('_exit') : true\r

    \                          ^\r

    \r

    TypeError: Cannot read property 'loaded' of undefined\r

    \    at exit (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97:27)\r

    \    at process.errorHandler (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:216:3)\r

    \    at process.emit (events.js:210:5)\r

    \    at processEmit [as emit]
    (C:/Users/xmr/Desktop/nyc/node_modules/signal-exit/index.js:155:32)\r

    \    at process._fatalException (internal/process/execution.js:150:25)\r

    Error: Failed to replace env in config: ${APPDATA}

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:415:13

    \    at String.replace (<anonymous>)

    \    at envReplace (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:411:12)

    \    at parseField (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:389:7)

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:330:24

    \    at Array.forEach (<anonymous>)

    \    at Conf.add (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:328:23)

    \    at ConfigChain.addString (C:/Program
    Files/nodejs/node_modules/npm/node_modules/config-chain/index.js:244:8)

    \    at Conf.<anonymous> (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:316:10)

    \    at C:/Program
    Files/nodejs/node_modules/npm/node_modules/graceful-fs/graceful-fs.js:115:16

    TypeError: Cannot read property 'loaded' of undefined

    \    at exit (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97:27)

    \    at errorHandler (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:216:3)

    \    at C:/Program Files/nodejs/node_modules/npm/bin/npm-cli.js:78:20

    \    at cb (C:/Program Files/nodejs/node_modules/npm/lib/npm.js:225:22)

    \    at C:/Program Files/nodejs/node_modules/npm/lib/npm.js:263:24

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:81:7

    \    at Array.forEach (<anonymous>)

    \    at Conf.<anonymous> (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:80:13)

    \    at Conf.f (C:/Program
    Files/nodejs/node_modules/npm/node_modules/once/once.js:25:25)

    \    at Conf.emit (events.js:210:5)

    C:/Program Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97\r

    \  var doExit = npm.config.loaded ? npm.config.get('_exit') : true\r

    \                          ^\r

    \r

    TypeError: Cannot read property 'loaded' of undefined\r

    \    at exit (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97:27)\r

    \    at process.errorHandler (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:216:3)\r

    \    at process.emit (events.js:210:5)\r

    \    at processEmit [as emit]
    (C:/Users/xmr/Desktop/nyc/node_modules/signal-exit/index.js:155:32)\r

    \    at process._fatalException (internal/process/execution.js:150:25)\r\n"
  wanted: ""
  compare: ===
  at:
    line: 12
    column: 5
    file: test/helpers/test-success.js
    function: testSuccess
  stack: |
    testSuccess (test/helpers/test-success.js:12:5)

 FAIL  test/nyc-integration.js
 ✖ stdout

  t.equal(stderr, '')
  t.matchSnapshot(stdout, 'stdout')
----^
}

  --- expected
  +++ actual
  @@ -1,11 +1,6 @@
  +----------|---------|----------|---------|---------|-------------------
  +File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
  +----------|---------|----------|---------|---------|-------------------
  +All files |       0 |        0 |       0 |       0 |
  +----------|---------|----------|---------|---------|-------------------
  -> @ test .
  -> node ./half-covered.js
  -
  ------------------|---------|----------|---------|---------|-------------------
  -File             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
  ------------------|---------|----------|---------|---------|-------------------
  -All files        |      50 |       50 |     100 |      50 |
  - half-covered.js |      50 |       50 |     100 |      50 | 6-8
  ------------------|---------|----------|---------|---------|-------------------
  -

  test: test/nyc-integration.js can run "npm test" which directly invokes a test file
  at:
    line: 13
    column: 5
    file: test/helpers/test-success.js
    function: testSuccess
  stack: |
    testSuccess (test/helpers/test-success.js:13:5)

 FAIL  test/nyc-integration.js
 ✖ should be equal


  t.equal(status, 0)
----^
  t.equal(stderr, '')
  t.matchSnapshot(stdout, 'stdout')

  test: test/nyc-integration.js can run "npm test" which indirectly invokes a test file
  found: 7
  wanted: 0
  compare: ===
  at:
    line: 11
    column: 5
    file: test/helpers/test-success.js
    function: testSuccess
  stack: |
    testSuccess (test/helpers/test-success.js:11:5)

 FAIL  test/nyc-integration.js
 ✖ should be equal

  t.equal(status, 0)
  t.equal(stderr, '')
----^
  t.matchSnapshot(stdout, 'stdout')
}

  test: test/nyc-integration.js can run "npm test" which indirectly invokes a test file
  found: "Error: Failed to replace env in config: ${APPDATA}

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:415:13

    \    at String.replace (<anonymous>)

    \    at envReplace (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:411:12)

    \    at parseField (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:389:7)

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:330:24

    \    at Array.forEach (<anonymous>)

    \    at Conf.add (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:328:23)

    \    at ConfigChain.addString (C:/Program
    Files/nodejs/node_modules/npm/node_modules/config-chain/index.js:244:8)

    \    at Conf.<anonymous> (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:316:10)

    \    at C:/Program
    Files/nodejs/node_modules/npm/node_modules/graceful-fs/graceful-fs.js:115:16

    TypeError: Cannot read property 'loaded' of undefined

    \    at exit (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97:27)

    \    at errorHandler (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:216:3)

    \    at C:/Program Files/nodejs/node_modules/npm/bin/npm-cli.js:78:20

    \    at cb (C:/Program Files/nodejs/node_modules/npm/lib/npm.js:225:22)

    \    at C:/Program Files/nodejs/node_modules/npm/lib/npm.js:263:24

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:81:7

    \    at Array.forEach (<anonymous>)

    \    at Conf.<anonymous> (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:80:13)

    \    at Conf.f (C:/Program
    Files/nodejs/node_modules/npm/node_modules/once/once.js:25:25)

    \    at Conf.emit (events.js:210:5)

    C:/Program Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97\r

    \  var doExit = npm.config.loaded ? npm.config.get('_exit') : true\r

    \                          ^\r

    \r

    TypeError: Cannot read property 'loaded' of undefined\r

    \    at exit (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97:27)\r

    \    at process.errorHandler (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:216:3)\r

    \    at process.emit (events.js:210:5)\r

    \    at processEmit [as emit]
    (C:/Users/xmr/Desktop/nyc/node_modules/signal-exit/index.js:155:32)\r

    \    at process._fatalException (internal/process/execution.js:150:25)\r

    Error: Failed to replace env in config: ${APPDATA}

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:415:13

    \    at String.replace (<anonymous>)

    \    at envReplace (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:411:12)

    \    at parseField (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:389:7)

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:330:24

    \    at Array.forEach (<anonymous>)

    \    at Conf.add (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:328:23)

    \    at ConfigChain.addString (C:/Program
    Files/nodejs/node_modules/npm/node_modules/config-chain/index.js:244:8)

    \    at Conf.<anonymous> (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:316:10)

    \    at C:/Program
    Files/nodejs/node_modules/npm/node_modules/graceful-fs/graceful-fs.js:115:16

    TypeError: Cannot read property 'loaded' of undefined

    \    at exit (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97:27)

    \    at errorHandler (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:216:3)

    \    at C:/Program Files/nodejs/node_modules/npm/bin/npm-cli.js:78:20

    \    at cb (C:/Program Files/nodejs/node_modules/npm/lib/npm.js:225:22)

    \    at C:/Program Files/nodejs/node_modules/npm/lib/npm.js:263:24

    \    at C:/Program Files/nodejs/node_modules/npm/lib/config/core.js:81:7

    \    at Array.forEach (<anonymous>)

    \    at Conf.<anonymous> (C:/Program
    Files/nodejs/node_modules/npm/lib/config/core.js:80:13)

    \    at Conf.f (C:/Program
    Files/nodejs/node_modules/npm/node_modules/once/once.js:25:25)

    \    at Conf.emit (events.js:210:5)

    C:/Program Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97\r

    \  var doExit = npm.config.loaded ? npm.config.get('_exit') : true\r

    \                          ^\r

    \r

    TypeError: Cannot read property 'loaded' of undefined\r

    \    at exit (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:97:27)\r

    \    at process.errorHandler (C:/Program
    Files/nodejs/node_modules/npm/lib/utils/error-handler.js:216:3)\r

    \    at process.emit (events.js:210:5)\r

    \    at processEmit [as emit]
    (C:/Users/xmr/Desktop/nyc/node_modules/signal-exit/index.js:155:32)\r

    \    at process._fatalException (internal/process/execution.js:150:25)\r\n"
  wanted: ""
  compare: ===
  at:
    line: 12
    column: 5
    file: test/helpers/test-success.js
    function: testSuccess
  stack: |
    testSuccess (test/helpers/test-success.js:12:5)

 FAIL  test/nyc-integration.js
 ✖ stdout

  t.equal(stderr, '')
  t.matchSnapshot(stdout, 'stdout')
----^
}

  --- expected
  +++ actual
  @@ -1,19 +1,6 @@
  +----------|---------|----------|---------|---------|-------------------
  +File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
  +----------|---------|----------|---------|---------|-------------------
  +All files |       0 |        0 |       0 |       0 |
  +----------|---------|----------|---------|---------|-------------------
  -> @ test .
  -> npm run test:deeper
  -
  -
  -> @ test:deeper .
  -> npm run test:even-deeper
  -
  -
  -> @ test:even-deeper .
  -> node ./half-covered.js
  -
  ------------------|---------|----------|---------|---------|-------------------
  -File             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
  ------------------|---------|----------|---------|---------|-------------------
  -All files        |      50 |       50 |     100 |      50 |
  - half-covered.js |      50 |       50 |     100 |      50 | 6-8
  ------------------|---------|----------|---------|---------|-------------------
  -

  test: test/nyc-integration.js can run "npm test" which indirectly invokes a test file
  at:
    line: 13
    column: 5
    file: test/helpers/test-success.js
    function: testSuccess
  stack: |
    testSuccess (test/helpers/test-success.js:13:5)

 FAIL  test/nyc-integration.js
 ✖ should be equal


  t.equal(status, 0)
----^
  t.equal(stderr, '')
  t.matchSnapshot(stdout, 'stdout')

  test: test/nyc-integration.js recursive run does not throw
  found: 1
  wanted: 0
  compare: ===
  at:
    line: 11
    column: 5
    file: test/helpers/test-success.js
    function: testSuccess
  stack: |
    testSuccess (test/helpers/test-success.js:11:5)
    runMicrotasks (<anonymous>)

 FAIL  test/nyc-integration.js
 ✖ should be equal

  t.equal(status, 0)
  t.equal(stderr, '')
----^
  t.matchSnapshot(stdout, 'stdout')
}

  test: test/nyc-integration.js recursive run does not throw
  found: "'true' is not recognized as an internal or external command,\r

    operable program or batch file.\r

    events.js:187\r

    \      throw er; // Unhandled 'error' event\r

    \      ^\r

    \r

    Error: spawn true ENOENT\r

    \    at notFoundError
    (C:/Users/xmr/Desktop/nyc/node_modules/cross-spawn/lib/enoent.js:6:26)\r

    \    at verifyENOENT
    (C:/Users/xmr/Desktop/nyc/node_modules/cross-spawn/lib/enoent.js:40:16)\r

    \    at ChildProcess.cp.emit
    (C:/Users/xmr/Desktop/nyc/node_modules/cross-spawn/lib/enoent.js:27:25)\r

    \    at Process.ChildProcess._handle.onexit
    (internal/child_process.js:272:12)\r

    Emitted 'error' event on ChildProcess instance at:\r

    \    at ChildProcess.cp.emit
    (C:/Users/xmr/Desktop/nyc/node_modules/cross-spawn/lib/enoent.js:30:37)\r

    \    at Process.ChildProcess._handle.onexit
    (internal/child_process.js:272:12) {\r

    \  code: 'ENOENT',\r

    \  errno: 'ENOENT',\r

    \  syscall: 'spawn true',\r

    \  path: 'true',\r

    \  spawnargs: []\r

    }\r\n"
  wanted: ""
  compare: ===
  at:
    line: 12
    column: 5
    file: test/helpers/test-success.js
    function: testSuccess
  stack: |
    testSuccess (test/helpers/test-success.js:12:5)
    runMicrotasks (<anonymous>)

 FAIL  test/nyc-integration.js 8 failed of 221 16s
 ✖ should be equal
 ✖ should be equal
 ✖ stdout
 ✖ should be equal
 ✖ should be equal
 ✖ stdout
 ✖ should be equal
 ✖ should be equal

 PASS  test/parser-plugins.js 6 OK 949.284ms
 PASS  test/process-args.js 5 OK 33.029ms
 PASS  test/processinfo.js 21 OK 2s
 PASS  test/report.js 3 OK 228.821ms
 PASS  test/should-instrument.js 23 OK 140.802ms
 PASS  test/source-map-support.js 2 OK 590.096ms
 PASS  test/temp-dir.js 15 OK 2s
 PASS  test/tsc.js 6 OK 2s
 PASS  test/wrap.js 12 OK 854.404ms


  🌈 SUMMARY RESULTS 🌈


 FAIL  test/nyc-integration.js 8 failed of 221 16s
 ✖ should be equal
 ✖ should be equal
 ✖ stdout
 ✖ should be equal
 ✖ should be equal
 ✖ stdout
 ✖ should be equal
 ✖ should be equal

Suites:   1 failed, 16 passed, 17 of 17 completed
Asserts:  8 failed, 411 passed, of 419
Time:     47s
npm ERR! Test failed.  See above for more details.
```

</details>

@coreyfarrell there you go :)

Might be worth looking into GitHub Actions CI (but the default shell is PorwerShell there, so I'm not sure if these will happen without switching to cmd)

--- a/test/helpers/run-nyc.js
+++ b/test/helpers/run-nyc.js
@@ -7,6 +7,12 @@ const envPath = {
   PATH: process.env.PATH
 }
 
+// Work around a Windows issue with `APPDATA`,
+// https://github.com/istanbuljs/nyc/issues/1248
+if ('APPDATA' in process.env) {
+  envPath.APPDATA = process.env.APPDATA
+}
+
 function sanitizeString (str, cwd, leavePathSep) {
   /*
    * File paths are different on different systems:

